README.md

```diff
@@ -2,31 +2,242 @@
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+In this diagram, the following hierarchy is represented:
 
-## Topic 2
+1. Classes that do not extend any other class explicitly will directly
+   inherit Java's `Object` class:
+    1. `String` extends `Object`.
+    2. `Number` extends `Object`.
+    3. `Person` extends `Object`.
+2. Classes that do extend other classes will then indirectly inherit from
+   Java's `Object` class:
+    1. `Integer` extends `Number`, which extends `Object`.
+    2. `Double` extends `Number`, which extends `Object`.
+    3. `BigDecimal` extends `Number`, which extends `Object`.
+    4. `Student` extends `Person`, which extends `Object`.
+    5. `Teacher` extends `Person`, which extends `Object`.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+_Note: Java's `Object` class is poorly named because an "object" is usually an
+instance of a class. But the class that all classes ultimately and implicitly
+extend from is called `Object`. In this course, as in most technical content, we
+use the term "object" to refer to the instance of a class. We will make sure the
+use the term "Java's Object class" when we are referring to the root class in
+Java.
+
+## Object Class Methods
+
+Since every class inherits from the `Object` class, this means that every class
+inherits the Java `Object` class' built-in methods. These are methods that can
+be used by any class or even overridden. Let's look at some of these methods
+and how they might be helpful:
+
+### `getClass()`
+
+The method `getClass()` will return the runtime class of the calling object.
+This means that the method will return the class that is evaluated at runtime.
+Here is an example using the `getClass()` method on an instance of `String` and
+an instance of `Bicycle`:
+
+```java
+public class Bicycle {
+   public static void main(String[] args) {
+      String greeting = "Hello World!";
+      System.out.println(greeting.getClass());
+      Bicycle bicycle = new Bicycle();
+      System.out.println(bicycle.getClass());
+   }
+}
+```
+
+The output of the above code would look like this:
+
+```text
+class java.lang.String
+class Bicycle
+```
+
+### `hashCode()`
+
+In computer science, there is a process called **hashing** which maps data to
+some integer value using the concept of hashing algorithms. Each object in Java
+is linked with some integer value that we call a **hash code**. Hashing helps
+us create distinct identifiers for objects in Java.
+
+Java's `Object` class has a method called `hashCode()` which allows us to get
+a hashed value of the object. It should be noted that the `hashCode()` method
+**does not return the memory address of the object**. The hashed value that the
```

Comment:
I might be misunderstanding the point, but the [JavaDocs](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode--) say this:

> As much as is reasonably practical, the hashCode method defined by class Object does return distinct integers for distinct objects. (This is typically implemented by converting the internal address of the object into an integer, but this implementation technique is not required by the Java™ programming language.)

Reply:
Java17 no longer includes a statement about internal address being used.  But I removed the sentence about it not being the memory address.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Object.html#hashCode()